server/accounts.go

```diff
@@ -832,12 +895,15 @@ func (s *Server) updateAccountClaims(a *Account, ac *jwt.AccountClaims) {
 	a.checkExpiration(ac.Claims())
 
 	// Clone to update, only select certain fields.
-	old := &Account{Name: a.Name, imports: a.imports, exports: a.exports, limits: a.limits}
+	old := &Account{Name: a.Name, imports: a.imports, exports: a.exports, limits: a.limits, signingKeys: a.signingKeys}
```

Comment:
Not your doing, but wonder if this line and the two below (where we reset the a.exports/a.imports) should not be protected with `a.mu.Lock()/Unlock()`. If we do so, and since it looks like updateSigningKeys is called only once here, I would put the code from that function here under the lock. Hope I am making sense.

Reply:
I believe you are correct.

Reply:
removed a function where I reset the signing keys and put it under the same lock

Reply:
Btw, regardless if it was a map or array, you need to do a copy here, otherwise you just have a pointer to the original map/array.

Reply:
As discussed, I understand that you want to "transfer" this to the "old" account since `a.signingKeys` will be set to nil and reconstructed below.